components/datetime/src/format/neo.rs

```diff
@@ -2217,13 +2213,21 @@ where
     ///     "The date is: November 20, 1700 Anno Domini"
     /// );
     /// ```
-    pub fn format_date<T>(&self, datetime: &'a T) -> FormattedDateTimePattern<'a>
+    pub fn format_date<I>(&self, datetime: &'a I) -> FormattedDateTimePattern<'a>
```

Comment:
This function is obsolete and should be deleted. Well, it should be replaced with a format function that binds on all of the input markers of `R`, like in NeoFormatter.

Reply:
How is this obsolete but `format_time` is not?

Reply:
I see a use case for these, you have a type that implements time fields, but you only want to format the date.

Reply:
The fields that get formatted are determined by the pattern you feed into this type, though.